src/RestServer/RestWebServer.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright (C) 2015-2023 The Neo Project.
+//
+// The Neo.Plugins.RestServer is free software distributed under the MIT software license,
+// see the accompanying file LICENSE in the main directory of the
+// project or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.HttpOverrides;
+using Microsoft.AspNetCore.ResponseCompression;
+using Microsoft.AspNetCore.Server.Kestrel.Https;
+using Microsoft.Extensions.DependencyInjection;
+using Neo.Plugins.RestServer.Middleware;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System.IO.Compression;
+using System.Net;
+using System.Net.Mime;
+using System.Net.Security;
+using System.Reflection;
+
+namespace Neo.Plugins.RestServer
+{
+    internal class RestWebServer
+    {
+        #region Globals
+
+        private readonly RestServerSettings _settings;
+
+        private IWebHost _host;
+
+        #endregion
+
+        public static bool IsRunning { get; private set; }
+
+        public RestWebServer(RestServerSettings settings)
+        {
+            _settings = settings;
+        }
+
+        public void Start()
+        {
+            if (IsRunning) return;
+
+            _host = new WebHostBuilder()
+                .UseKestrel(options =>
+                {
+                    // Web server configuration
+                    options.AddServerHeader = false;
+                    options.Limits.MaxConcurrentConnections = _settings.MaxConcurrentConnections;
+                    options.Limits.KeepAliveTimeout = TimeSpan.FromMinutes(_settings.KeepAliveTimeout);
+                    options.Limits.RequestHeadersTimeout = TimeSpan.FromSeconds(15);
+                    options.Listen(_settings.BindAddress, unchecked((int)_settings.Port), listenOptions =>
+                    {
+                        if (string.IsNullOrEmpty(_settings.SslCertFile)) return;
+                        listenOptions.UseHttps(_settings.SslCertFile, _settings.SslCertPassword, httpsOptions =>
```

Comment:
don't understand why we need this for https. If cert is valid their shouldn't be a problem, I thought. Copied from RPC Server.